src/compiler/crystal/codegen/codegen.cr

```diff
@@ -1864,6 +1891,18 @@ module Crystal
       builder.unreachable
     end
 
+    def debug_compiler_log(file = __FILE__, line = __LINE__)
+      return unless ENV["CRYSTAL_DEBUG_COMPILER"]?
+      msg = yield || ""
+      puts("<#{Crystal.relative_filename(file)}:#{line}> #{msg}")
+      nil
+    end
+
+    # :ditto:
+    def debug_compiler_log(file = __FILE__, line = __LINE__)
+      debug_compiler_log(file, line) { }
+    end
+
```

Comment:
If I'm not mistaken this is used in just one place, is worth the addition then?

Reply:
It was used for testing. It still may be useful later on when debug needs to be done in case if debug itself cannot show that information.